src/otx/algorithms/detection/adapters/mmdet/configurer.py

```diff
@@ -149,6 +150,13 @@ def is_mmov_model(key, value):
                 {"model_path": ir_model_path, "weight_path": ir_weight_path, "init_weight": ir_weight_init},
             )
 
+        # Test config
+        if max_num_detections > 0:
+            logger.info(f"Model max_num_detections: {max_num_detections}")
+            test_cfg = cfg.model.test_cfg
+            test_cfg.max_per_img = max_num_detections
+            test_cfg.nms_pre = max_num_detections * 10
+
```

Comment:
Two stage detector, such as maskrcnn, has parameters to control rpn output. Shouldn't we change that parameters too?

Reply:
Hmm, I forgot them. Thanks for the catch.

Reply:
Addressed in https://github.com/openvinotoolkit/training_extensions/pull/2647/commits/1627d4c0a064ebc4144ebb354c1e6226969e0dcd

Reply:
If max_dets param is only for inference then, we don't need

Reply:
Well... I've been also thinking about it; whether it would have been affecting the loss flow?
Anyway, let's be consistent at least. There might be edge cases, but currently those are all we have regarding the `max_per_img`.
https://github.com/openvinotoolkit/training_extensions/pull/2647/commits/5979c3f94ad96c17d7058abf4728e735ab104240